src/openforms/contrib/zgw/service.py

```diff
@@ -193,7 +193,7 @@ def create_rol(zaak: dict, initiator: dict, options: dict) -> Optional[dict]:
         "zaak": zaak["url"],
         # "betrokkene": initiator.get("betrokkene", ""),
         "betrokkeneType": initiator.get("betrokkeneType", "natuurlijk_persoon"),
-        "roltype": rol_typen["results"][0]["url"],
+        "roltype": initiator.get("roltype", rol_typen["results"][0]["url"]),
```

Comment:
maybe only fetch `rol_typen` here if there's no `roltype` provided, now we're doing calls to APIs that only slows down things while the result isn't even used.

I'd also rename `initiator` to `betrokkene` here, since the same function is now used in different (non-initiator) contexts.

The logic to find the list of `roltypen` inside of this function should also make use of the new `retrieve_roltype` function - possibly you can refactor that to make the API calls and then provide a `matcher` callback to extract the relevant `roltype`. This keeps things DRY.